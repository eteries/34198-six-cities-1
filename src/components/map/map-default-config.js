export const MapDefaultConfig = {
  SOURCE_URL: `https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`,
  SOURCE_OPTIONS: {
    attribution: `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`
  },
  ICON: {
    iconUrl: `img/pin.svg`,
    iconSize: [30, 30]
  },
  ZOOM_CONTROL: false,
  SHOW_MARKER: true,
  OPTIONS: {
    center: [0, 0],
    zoomControl: false,
    marker: true,
    zoom: 12
  }
};
