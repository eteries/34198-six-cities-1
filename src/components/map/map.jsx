import React, {PureComponent} from 'react';
import leaflet from 'leaflet';

import {MapDefaultConfig as Config} from './map-default-config';

export class Map extends PureComponent {
  constructor(props) {
    super(props);
    this.map = null;
  }

  init() {
    const city = [52.38333, 4.9];
    this.map = leaflet.map(`map`, Object.assign({center: city}, Config.OPTIONS));
    this.map.setView(city, Config.OPTIONS.zoom);

    leaflet
      .tileLayer(Config.SOURCE_URL, Config.SOURCE_OPTIONS)
      .addTo(this.map);
  }

  addMarkers() {
    const offerCords = [52.3709553943508, 4.89309666406198];

    leaflet
      .marker(offerCords, {icon: leaflet.icon(Config.ICON)})
      .addTo(this.map);
  }

  componentDidMount() {
    this.init();
    this.addMarkers();
  }

  componentWillUnmount() {
    this.map = null;
  }

  render() {
    return <section className="cities__map map" id="map" />;
  }
}

