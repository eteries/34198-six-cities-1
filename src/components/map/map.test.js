import React from 'react';
import renderer from 'react-test-renderer';

import {Map} from './map.jsx';

/**
 * @jest-environment jsdom
 */

describe(`Map works`, () => {
  it(`Maps renders`, () => {
    const div = document.createElement(`div`);
    div.setAttribute(`id`, `map`);
    document.body.appendChild(div);

    const layout = renderer
      .create(<Map />)
      .toJSON();

    expect(layout).toMatchSnapshot();
  });
});
