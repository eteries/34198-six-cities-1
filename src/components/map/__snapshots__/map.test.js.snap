// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map works Maps renders 1`] = `
<section
  className="cities__map map"
  id="map"
/>
`;
